pkg/apiserver/user/auth.go

```diff
@@ -224,7 +224,12 @@ func (s *AuthService) authSQLForm(f *authenticateForm) (*utils.SessionUser, erro
 		// tidb.ErrTiDBAuthFailed
 		return nil, err
 	}
-	defer db.Close() //nolint:errcheck
+	defer func() {
+		sqlDB, err := db.DB()
+		if err == nil {
+			sqlDB.Close()
+		}
+	}()
```

Comment:
Why do we need to change to this? (I didn't find this change in [gorm v2 release note](https://gorm.io/docs/v2_release_note.html))

Reply:
Cuz there's no `Close` interface in v2 `*gorm.DB`. https://pkg.go.dev/gorm.io/gorm#pkg-index

And, is it really necessary to close the connection pool?
https://github.com/go-gorm/gorm/issues/3834
https://github.com/go-gorm/gorm/issues/3535
https://github.com/go-gorm/gorm/issues/4157